Composer/packages/lib/code-editor/src/LuEditor.tsx

```diff
@@ -2,12 +2,131 @@
 // Licensed under the MIT License.
 
 import React from 'react';
+import { listen, MessageConnection } from 'vscode-ws-jsonrpc';
+import * as monacoEditor from '@bfcomposer/monaco-editor/esm/vs/editor/editor.api';
+import * as monacoCore from 'monaco-editor-core';
+import get from 'lodash/get';
+import { MonacoServices, MonacoLanguageClient } from 'monaco-languageclient';
 
+import { registerLULanguage } from './languages';
+import { createUrl, createWebSocket, createLanguageClient } from './utils/lspUtil';
 import { RichEditor, RichEditorProps } from './RichEditor';
 
-const LU_HELP = 'https://aka.ms/lu-file-format';
-const placeholder = `> See ${LU_HELP} to learn about supported LU concepts.`;
+const LU_HELP = 'https://github.com/microsoft/botframework-cli/blob/master/packages/lu/docs/lu-file-format.md';
+const placeholder = `> To learn more about the LU file format, read the documentation at
+> ${LU_HELP}`;
 
-export function LuEditor(props: RichEditorProps) {
-  return <RichEditor placeholder={placeholder} helpURL={LU_HELP} {...props} />;
+export interface LUOption {
+  inline: boolean;
+  content: string;
+  template?: {
+    name: string;
+    parameters?: string[];
+    body: string;
+  };
+}
+
+export interface LULSPEditorProps extends RichEditorProps {
+  luOption?: LUOption;
+  languageServer?:
+    | {
+        host?: string;
+        hostname?: string;
+        port?: number | string;
+        path: string;
+      }
+    | string;
+}
+
+const defaultLUServer = {
+  path: '/lu-language-server',
+};
+declare global {
+  interface Window {
+    monacoServiceInstance: MonacoServices;
+    monacoLUEditorInstance: MonacoLanguageClient;
+  }
+}
+
+function convertEdit(serverEdit: any) {
```

Comment:
The names used here are unclear. `convertEdit` to what? A transform is happening but I don't know the beginning and ending shape. Adding a real type as the `serverEdit` parameter would be helpful here.